lib/glific/stats.ex

```diff
@@ -449,4 +449,25 @@ defmodule Glific.Stats do
     })
     |> Repo.one()
   end
+
+  @doc """
+  Get the details of the usage for this organization, from start_date to end_date both inclusive
+  """
+  @spec get_bsp_message_count(non_neg_integer()) :: {:ok, non_neg_integer()}
+  def get_bsp_message_count(organization_id) do
+    start_date = Timex.shift(DateTime.utc_now(), days: -1)
+    end_date = DateTime.utc_now()
+
+    %{messages: messages} =
+      Stat
+      |> where([s], s.organization_id == ^organization_id)
+      |> where([s], s.period == "hour")
+      |> where([s], s.inserted_at > ^start_date and s.inserted_at <= ^end_date)
+      |> select([s], %{
```

Comment:
I don't think get_bsp_message_count name makes sense here.

get_message_count(org_id, period, opts) 

you can send start date and end date in the opts itself.